### ISSUE TITLE
Regex: should warn on negative fixed quantifier

### ISSUE THREAD

**Issue by chborl** [dotnet/roslyn#32995]
```csharp
namespace ConsoleApp27
{
    class Program
    {
        static void Main(string[] args)
        {
            //language=regex
            string s = @"a{-3,4}";
        }
    }
}
```

Expected:
Diagnostic: negative numbers not allowed in fixed quantifier

Actual:
No diagnostic

**Comment by CyrusNajmabadi** [comment]
By design. The .net regex engine does not error here.  Our goal is to match the actual engine that people will be executing against.  We don't deviate except in cases where it was too exceptionally complex to try to match their behavior

**Comment by CyrusNajmabadi** [comment]
Note: this is not actually an error. This is simply not a quantifier.  Try this program out:

```c#
            var regex = new Regex("a{-3,4}");
            Console.WriteLine(regex.IsMatch("a{-3,4}"));
```

This prints true.  We should not error on legal regexes that the underlying engine supports.

**Comment by sharwell** [comment]
I verified that syntax highlighting is also correct for this case.
